src/compiler/core.ts

```diff
@@ -2250,18 +2250,40 @@ namespace ts {
         }
     }
 
-    export function padLeft(s: string, length: number) {
-        while (s.length < length) {
-            s = " " + s;
+
+    /**
+     * Returns 's' left-padded with copies of 'padString' until it reaches the given 'length'.
+     * If omitted, 'padString' defaults to a single space.
+     * If 'length' is less than or equal to 's.length' or 'padString' is empty, returns 's' unchanged.
+     */
+    export function padLeft(s: string, length: number, padString = " ") {

```

Comment:
Personally, I would probably use a simpler, less general implementation for an internal helper like this.  For example, you could declare `padString: " " | "0" = " "` and then you wouldn't need to deal with length checks on `padString`.  In the unlikely event that someone needs a different padding character, it'll be obvious to them how to proceed.

Reply:
I hear what you are saying, no problem. Kinda why I wanted to get your thoughts before banging this around any more!